rts/lib/libcpuid/libcpuid/cpuid_main.c

```diff
@@ -290,7 +290,12 @@ static bool set_cpu_affinity(logical_cpu_t logical_cpu)
 
 static int get_total_cpus(void)
 {
+#ifdef __OpenBSD__
+	// HW_NCPUONLINE accounts for hyperthreading off/on
+	int mib[2] = { CTL_HW, HW_NCPUONLINE };
+#else
 	int mib[2] = { CTL_HW, HW_NCPU };
```

Comment:
perhaps regular linux could use the new constant as well?

Reply:
Linux seems to use something similar judging from the variable name `_SC_NPROCESSORS_ONLN`, see https://github.com/beyond-all-reason/spring/blob/BAR105/rts/lib/libcpuid/libcpuid/cpuid_main.c#L253. This segment with `HW_NCPU` is for various *BSD OS, QNX. Just looking at [FreeBSD's sysctl(3) man page](https://man.freebsd.org/cgi/man.cgi?sysctl(3)), they don't seem to have this kind of MIB entry yet.